### ISSUE TITLE
Merge with obmenu2?

### ISSUE THREAD

**Issue by 0x10** [keithbowes/obmenu#2]
Sorry for misusing the issue system for messaging.

Without being aware about the effort you took to port obmenu, I too took the effort.
But I didn't port it using 2to3 but rewrote it from scratch, to get rid of every deprecation.
Link: https://github.com/0x10/obmenu2

I acknowledge your effort and your future plans to make it less openbox specific. The codebase you are currently maintaining still contains deprecated use of interfaces (e.g. GTK STOCK_* enums).

I don't know about Waybox-specific features. So I would offer the following:
What do you think about merging both projects into one and than hand it over to you to add any additional waybox features?!

**Comment by keithbowes** [comment]
Yeah, I haven't gotten around to writing a menu system in Waybox yet, but it'll come, hopefully when I get support for wlr-layer-shell implemented, writing a menu system will be fairly easy.  I tried replacing everything glade said was deprecated, but I'll look into getting rid of deprecated features.  I'm also planning on looking into porting some of the features from obmenux over, so doing the same for openbox2 shouldn't be a problem..

The only thing I have planned for the Waybox menu is optional Vi keys, like I was able to use in WindowMaker, before going to Fluxbox, then to Openbox, and now having given up the legacy X.

**Comment by keithbowes** [comment]
> Without being aware about the effort you took to port obmenu, I too took the effort. But I didn't port it using 2to3 but rewrote it from scratch, to get rid of every deprecation. Link: https://github.com/0x10/obmenu2
> 
> I acknowledge your effort and your future plans to make it less openbox specific. The codebase you are currently maintaining still contains deprecated use of interfaces (e.g. GTK STOCK_* enums).
>

Yes, it takes a lot of effort to get rid of every deprecation, as my recent attempt to get rid of the deprecated GTK 4 features. Replacing GtkComboBox with GtkDropDown was relatively easy, but replacing GtkTreeView with GtkColumnView was a major undertaking and is what made me nix GTK 3 support (the entire tree view system would essentially require two versions: one for GTK 3 and one for GTK 4). That's only in the gtk4 branch but the version in the master branch still works in GTK 3 (and in GTK 4, with warnings about Gtk.TreeView being deprecated). However, I'm thinking of merging gtk4 into master soon; GTK 4 is over five years old (and the GTK 4.12 required in the gtk4 branch is over two years old), so it's not just freshly released like it was when I initially added GTK 4 support.

I'm not sure what there is in obmenu2 to merge. I'm thinking of some features, like support for item icons, the ability to work with multiple menu files at once, etc., but I haven't done it yet.

I know it's been years since this issue was filed but I was just looking at the issues to see if there's anything I need to fix while doing work on the gtk4 branch before pushing it (git says that my local copy is 18 commits ahead of the one on GitHub).
